src/main/java/uk/gov/hmcts/reform/wacaseeventhandler/services/EventMessageReceiverService.java

```diff
@@ -246,4 +246,27 @@ private boolean validate(String messageId, EventInformation eventInformation, Bo
             && isNotBlank(eventInformation.getEventTimeStamp().toString())
             && fromDlq != null;
     }
+
+    public List<String> getUnprocessableMessages() {
+        log.info("Retrieving Unprocessable messages from case db");
+        List<String> messageIds = repository.getUnprocessableMessages();
+        log.info("Retrieving Unprocessable messages from case db {}", messageIds);
+        return messageIds;
+    }
+
+    public List<String> getReadyMessages() {
+        log.info("Retrieving ready messages from case db");
+        List<String> messageIds = repository.getReadyMessages();
+        log.info("Retrieving ready messages from case db '{}'", messageIds);
+        return messageIds;
+    }
+
+    public List<String> findProblematicMessages(MessageState messageState) {
+        log.info("Retrieving problematic messages of type {} from case db", messageState.name());
+        List<String> messageIds = repository.findProblematicMessages(messageState);
+        log.info("Retrieved problematic message id '{}' from case db", messageIds);
+        return messageIds;
+    }
+
```

Comment:
Not sure this code really belongs in this class?  `EventMessageReceiverService` sounds like it is about receiving messages whereas here we are executing a sort of admin job.

Reply:
done